app/main.py

```diff
@@ -2,0 +1,37 @@
+class Animal:
+    animal = []
+    
+    def __init__(self, name, hidden=False, health=100):
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+        
+    def health_stats(self):
+        if (self.health <= 0):
+            self.health = 0
+            Animal.remove_dead()
+
+    @classmethod
+    def remove_dead(cls):
+        cls.alive = [animal for animal in cls.alive if animal.health == 0]
```

Comment:
The logic in `remove_dead` is incorrect. It should retain animals with health greater than 0. Change the condition to `animal.health > 0`.